types/index.d.ts

```diff
@@ -779,10 +779,10 @@ declare class Realm {
     removeAllListeners(name?: string): void;
 
     /**
-     * @param  {()=>void} callback
-     * @returns void
+     * @param  {()=>ReturnValueType} callback
+     * @returns {ReturnValueType}
      */
-    write(callback: () => void): void;
+    write<ReturnValueType>(callback: () => ReturnValueType): ReturnValueType;
```

Comment:
Does this `ReturnValueType` need a default value or is it able to infer this entirely from the callback? I would have thought you needed to use the `infer` keyword or `ReturnType` helper type or at least `write<ReturnValueType = unknown>(...)`.

Reply:
I worked on this with José.  The type of the return value is inferred from the callback return value (we tested this).  That being said, the default value should probably be set to unknown (the value is just typed as `void` if nothing is returned).

Reply:
I just tested it in the TS playground - it works as I expect it to. Good job!
https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwHcYsMQAeAJQD4AKMKCCAIyjAGsAueGgSngF4q8Cjy4UA3AChJGAJ4AHBAFUAziBgD4Ab0nw98VFAC2ILiozFUAcykBfaWDzn4yNRv6FipGrwFCd+vCOqM6u6lyq6poBgfqGJlwARABSOAAWqIkANLr6tlKBcBjIMPhhMHY8UpLBKjgQIAB0EDhWNOWN8SBVkkA